networking/src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -339,16 +344,24 @@ setInFlightFor Packet{..} f var = liftIO $ modifyMVar_ var (return . update)
     updateInnerMap :: Map Precedence Int -> Map Precedence Int
     updateInnerMap = Map.adjust f packetPrec
 
+-- | Connection status
+data ConnectionStatus =
+      Connecting
+    | Connected
+    | ConnectionBroken
```

Comment:
Is this one of any use? Isn't it always `NotConnected -> Connecting -> Connected -> Recurse` ? What would be the policy here? i.e. how long would the status remain "broken" before it becomes `NotConnected` or `Connecting`?

Reply:
It would stay broken until a request is send in which case it will change into `Connecting` -  how long it takes depends on the send policy / node being rate limited.  If there was just an error an the policy is `SendAll` the two values `ConnectionBroken` and `Connecting` will flicker.   Another strategy is to skip the `Connecting` and keep it `ConnectionBroken` until we get `Connected`.